aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,327 @@
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/SampleLoggingActionFilterAttribute.cs
+  :language: c#
+  :emphasize-lines: 1, 3, 7
+  :lines: 8-38
+  :dedent: 4
+
+This filter can be applied to classes or methods using the ``[SampleLoggingActionFilter]`` syntax.
+
+``IFilterFactory`` implements ``IFilter``. Therefore, an ``IFilterFactory`` instance can be used as an ``IFilter`` instance anywhere in the filter pipeline. When the framework prepares to invoke the filter, attempts to cast it to an ``IFilterFactory``. If that cast succeeds, the ``CreateInstance`` method is called to create the ``IFilter`` instance that will be invoked. This provides a very flexible design, since the precise filter pipeline does not need to be set explicitly when the application starts.
+
+You can implement ``IFilterFactory`` on your own attribute implementations as another approach to creating filters:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/AddHeaderWithFactoryAttribute.cs
+  :language: c#
+  :emphasize-lines: 1,4-7
+  :lines: 6-26
+  :dedent: 4
+
+.. _ordering:
+
+Ordering
+^^^^^^^^
+Filters can be applied to action methods or controllers (via attribute) or added to the global filters collection. Scope also generally determines ordering. The filter closest to the action runs first; generally you get overriding behavior without having to explicitly set ordering. This is sometimes referred to as "Russian doll" nesting, as each increase in scope is wrapped around the previous scope, like a `nesting doll <https://en.wikipedia.org/wiki/Matryoshka_doll>`_.
+
+In addition to scope, filters can override their sequence of execution by implementing `IOrderedFilter <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Filters/IOrderedFilter/index.html>`_. This interface simply exposes an ``int`` ``Order`` property, and filters execute in ascending numeric order based on this property. All of the built-in filters, including ``TypeFilterAttribute`` and ``ServiceFilterAttribute``, implement ``IOrderedFilter``, so you can specify the order of filters when you apply the attribute to a class or method. By default, the ``Order`` property is 0 for all of the built-in filters, so scope is used as a tie-breaker and (unless ``Order`` is set to a non-zero value) is the determining factor.
+
+To see the scope-based ordering in action, consider the following class, which has the same filter applied at both the class and method level. The same filter is also registered globally. When run, the log output will show the order in which the filters executed, and how these compared to the ``Controller`` base class's ``OnActionExecuting`` and ``OnActionExecuted`` methods.
+
+.. literalinclude:: filters/sample/src/FiltersSample/Controllers/OrderingController.cs
```

Comment:
AFAIK it doesn't log anything when a particular filter executes. Unless this is changed in RC2, the logging statements are necessary if you want to see the order in which your filters are applied. Here is all of the logging output for the sample shown in this part of the article:
![image](https://cloud.githubusercontent.com/assets/782127/14149383/f76a2768-f671-11e5-9089-320e958d242d.png)

Green arrows are my logging statements; blue arrows are built-in framework logging.